pandas/core/groupby/groupby.py

```diff
@@ -1275,8 +1275,14 @@ def std(self, ddof: int = 1):
         Series or DataFrame
             Standard deviation of values within each group.
         """
-        # TODO: implement at Cython level?
-        return np.sqrt(self.var(ddof=ddof))
+        result = self.var(ddof=ddof)
+        if result.ndim == 1:
+            result = np.sqrt(result)
+        else:
+            for col in result:
+                if col not in self.exclusions:
```

Comment:
rather than doing like this

```
cols = result.columns.difference(self.exclusions)
result[cols] = np.sqrt(result[cols])
```

Reply:
I like this a lot - but duplicate column names and non-unique multi-index are both causing issues. The test that is failing is groupby.test_groupby.test_dup_labels_output_shape. I also tried

    cols = result.columns.difference(self.exclusions)
    roots = np.sqrt(result[cols])
    for col in cols:
        result[col] = roots[col]

which gets around the duplicate column name issue while avoiding multiple calls to np.sqrt, but unfortunately still succumbs to non-unique multi-index.

Reply:
you can use .iloc to do this instead (so get column positions; use `.get_indexer`)

Reply:
I am currently trying:

    cols = result.columns.get_indexer_for(
        result.columns.difference(self.exclusions).unique()
    )
    result.iloc[:, cols] = np.sqrt(result.iloc[:, cols])

Is this what you had in mind? It still does not work when columns are not unique (both with and without .unique() there). I believe the issue is iLocIndexer._setitem_with_indexer whose comment reads:

    This method is currently broken when dealing with non-unique Indexes,
    since it goes from positional indexers back to labels when calling
    BlockManager methods, see GH#12991, GH#22046, GH#15686.

Reply:
Ah, I think I have a solution - use .values on the end of the last line. Pushing shortly.